### PR TITLE
Wait for a service to be provisioned

### PR DESCRIPTION
**Pull request by agrare**
Wait for the service_instance to be provisioned before marking the task complete

Depends on:

- [x] https://github.com/ManageIQ/topological_inventory-collector-openshift/pull/42
- [x] https://github.com/ManageIQ/topological_inventory-api/pull/148

### CODE REVIEW

File: lib/topological_inventory/openshift/operations/core/service_catalog_client.rb [hunk 1]
```diff
@@ -51,6 +63,29 @@ def build_payload(service_plan_name, service_offering_name, order_parameters)
...
+
+              break unless service_instance_provisioning?(service_instance)
+            end
+
+            service_instance
+          end
+
+          def service_instance_provisioning?(service_instance)
+            reason = service_instance.status.conditions.first&.reason
+            %w(Provisioning ProvisionRequestInFlight).include?(reason)
```
Comment (agrare) [inline lib/topological_inventory/openshift/operations/core/service_catalog_client.rb]: @mkanoor I got these from a script you sent me a while ago, but I'm also seeing `OrphanMitigationComplete` have you seen this state?  It seems to flip back and forth between that and `ProvisionRequestInFlight`

Comment (agrare) [reply lib/topological_inventory/openshift/operations/core/service_catalog_client.rb]: Might be related to https://github.com/kubernetes-incubator/service-catalog/issues/2514

File: lib/topological_inventory/openshift/operations/core/service_catalog_client.rb [hunk 2]
```diff
@@ -1,64 +1,93 @@
...
+              :spec       => {
+                :clusterServiceClassExternalName => service_offering_name,
+                :clusterServicePlanExternalName  => service_plan_name,
+                :parameters                      => safe_params
+              }
             }
+          end
+
+          def wait_for_provision_complete(connection, service_instance)
+            loop do
```
Comment (Ladas) [inline lib/topological_inventory/openshift/operations/core/service_catalog_client.rb]: So we'll need to do sync waiting, until we have some tool for async workers, right?

File: lib/topological_inventory/openshift/operations/core/service_catalog_client.rb (hunk 1 above)
Comment (Ladas) [reply lib/topological_inventory/openshift/operations/core/service_catalog_client.rb]: hm, might be safer to check for finished and error states?

Comment (agrare) [reply lib/topological_inventory/openshift/operations/core/service_catalog_client.rb]: Yeah that's a good idea, hopefully there are fewer of those :)

**Comment by agrare** [comment]
@Ladas rebased

**Pull request merged**